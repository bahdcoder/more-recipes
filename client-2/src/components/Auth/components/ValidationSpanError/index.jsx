import React from 'react';
import PropTypes from 'prop-types';

const ValidationSpanError = ({ error }) => (
  <span className="validation-span-error">
    <small style={{
      color: '#E27C3E'
    }}
    >
      {error}
    </small>
  </span>
);

ValidationSpanError.propTypes = {
  error: PropTypes.string.isRequired
};

export default ValidationSpanError;
