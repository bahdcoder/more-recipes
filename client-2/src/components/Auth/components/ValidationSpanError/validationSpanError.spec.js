import React from 'react';
import renderer from 'react-test-renderer';

import ValidationSpanError from './index';

describe('The ValidationSpanError', () => {
  it('Should render an error in a span', () => {
    const error = 'The email is required.';
    const tree = renderer.create(<ValidationSpanError error={error} />).toJSON();

    expect(tree.type).toEqual('span');
    expect(tree.children.length).toEqual(1);
    expect(tree.children[0].type).toEqual('small');
    expect(tree.children[0].children.length).toEqual(1);
    expect(tree.children[0].children[0]).toEqual(error);
    expect(tree).toMatchSnapshot();
  });
});
