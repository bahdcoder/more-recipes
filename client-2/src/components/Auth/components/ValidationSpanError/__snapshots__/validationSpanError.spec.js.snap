// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ValidationSpanError Should render an error in a span 1`] = `
<span
  className="validation-span-error"
>
  <small
    style={
      Object {
        "color": "#E27C3E",
      }
    }
  >
    The email is required.
  </small>
</span>
`;
